### PR TITLE
#1289 change expense type to autocomplete

### LINKED ISSUE #1289: Finance Fixes - Change Expense Type to be an Autocomplete
## Description

There are roughly 20 expense types on production that are options, so it makes for sense for the Expense Field in the reimbursement request form to be an autocomplete select rather than just a select.

## Acceptance Criteria

- The select on the create/edit reimbursement request form is now an autocomplete select.
- The create and edit forms functionality is unaffected

### PR DESCRIPTION
**Pull request by haderie**
## Changes

the create/edit reimbursement request form is now an autocomplete select
The create and edit forms functionality should be unaffected
NERAutocomplete now looks like NativeSelect
## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

expenseTypeSmall
expenseTypeLarge

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1289

### CODE REVIEW

**Review by ryanhowe28 — APPROVED** [review]
Looks great!

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx [hunk 1]
```diff
@@ -188,17 +189,17 @@ const ReimbursementRequestFormView: React.FC<ReimbursementRequestFormViewProps>
...
-                      value={value}
-                      error={!!errors.expenseTypeId}
-                    >
-                      {allExpenseTypes.map((expenseType) => (
-                        <MenuItem key={expenseType.expenseTypeId} value={expenseType.expenseTypeId}>
-                          {expenseType.name}
-                        </MenuItem>
-                      ))}
-                    </Select>
+                    <Autocomplete
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx:192]: ok so, you should use the NERAutocomplete component, but like lowkey, I like the way this one looks better 💀 . If you're cool with this ticket getting a lil more involved, could you change the NERAutocomplete to look like this then use it in this spot instead

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by ryanhowe28 — COMMENTED** [review]
Also try and see how it looks without the double "expense type" words when selecting. Don't know if the smaller red one is necessary.

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx [hunk 2]
```diff
@@ -187,19 +193,28 @@ const ReimbursementRequestFormView: React.FC<ReimbursementRequestFormViewProps>
...
-                        <MenuItem key={expenseType.expenseTypeId} value={expenseType.expenseTypeId}>
-                          {expenseType.name}
-                        </MenuItem>
-                      ))}
-                    </Select>
-                  )}
+                  render={({ field: { onChange, value } }) => {
+                    return (
+                      <NERAutocomplete
+                        options={allExpenseTypes.map(expenseTypesToAutocomplete)}
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx:199]: because you call `allExpenseTypes.map(expenseTypesToAutocomplete)`, can you make it a variable and then call it in both places you use it?

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx [hunk 3]
```diff
@@ -187,19 +193,28 @@ const ReimbursementRequestFormView: React.FC<ReimbursementRequestFormViewProps>
...
+                        value={
+                          allExpenseTypes
+                            .map(expenseTypesToAutocomplete)
+                            .find((expenseType) => expenseType.label === value) || null
+                        }
+                        placeholder="Expense Type"
+                        onChange={(_event, newValue) => {
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx:206]: you can use a ternary operator inside a call of onChange to simplify this method.

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx [hunk 4]
```diff
@@ -187,19 +193,28 @@ const ReimbursementRequestFormView: React.FC<ReimbursementRequestFormViewProps>
...
+                          if (newValue) {
+                            onChange(newValue.label);
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx:208]: also you'll wanna use .id, not .label here

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx [hunk 5]
```diff
@@ -100,6 +102,10 @@ const ReimbursementRequestFormView: React.FC<ReimbursementRequestFormViewProps>
     </FormControl>
   );
 
+  const expenseTypesToAutocomplete = (expense: ExpenseType): { label: string; id: string } => {
+    return { label: expense.name, id: expense.toString() };
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx:106]: should be expense.expenseTypeId

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx [hunk 6]
```diff
@@ -100,6 +102,10 @@ const ReimbursementRequestFormView: React.FC<ReimbursementRequestFormViewProps>
     </FormControl>
   );
 
+  const expenseTypesToAutocomplete = (expense: ExpenseType): { label: string; id: string } => {
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx:105]: this variable name should be expenseType, not expense

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx [hunk 7]
```diff
@@ -187,19 +193,28 @@ const ReimbursementRequestFormView: React.FC<ReimbursementRequestFormViewProps>
...
+                        placeholder="Expense Type"
+                        onChange={(_event, newValue) => {
+                          if (newValue) {
+                            onChange(newValue.label);
+                          } else {
+                            onChange('');
+                          }
+                        }}
+                        id={'expenseType'}
+                        size={undefined}
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx:214]: set this to small/medium and see how it looks, but I wouldn't set it as undefined like this

**Review by RChandler234 — CHANGES_REQUESTED** [review]
Could you also make the NERAutocomplete look like the autocomplete with the native select you were using originally? I think it fits a lot better with the components we usually use than the way NERAutocomplete looks now

**Review by ryanhowe28 — DISMISSED** [review]
Looks great! I think all the other auto completes look like they fit much better.

**Review by RChandler234 — APPROVED** [review]
